code/modules/mod/modules/modules_supply.dm

```diff
@@ -0,0 +1,577 @@
+//Supply modules for MODsuits
+
+///Internal GPS - Extends a GPS you can use.
+/obj/item/mod/module/gps
+	name = "MOD internal GPS module"
+	desc = "This module uses common Nanotrasen technology to calculate the user's position anywhere in space, \
+		down to the exact coordinates. This information is fed to a central database viewable from the device itself, \
+		though using it to help people is up to you."
+	icon_state = "gps"
+	module_type = MODULE_ACTIVE
+	complexity = 1
+	use_power_cost = DEFAULT_CHARGE_DRAIN * 0.2
+	incompatible_modules = list(/obj/item/mod/module/gps)
+	cooldown_time = 0.5 SECONDS
+	device = /obj/item/gps/mod
+
+///Hydraulic Clamp - Lets you pick up and drop crates.
+/obj/item/mod/module/clamp
+	name = "MOD hydraulic clamp module"
+	desc = "A series of actuators installed into both arms of the suit, boasting a lifting capacity of almost a ton. \
+		However, this design has been locked by Nanotrasen to be primarily utilized for lifting various crates. \
+		A lot of people would say that loading cargo is a dull job, but you could not disagree more."
+	icon_state = "clamp"
+	module_type = MODULE_ACTIVE
+	complexity = 3
+	use_power_cost = DEFAULT_CHARGE_DRAIN
+	incompatible_modules = list(/obj/item/mod/module/clamp)
+	cooldown_time = 0.5 SECONDS
+	overlay_state_inactive = "module_clamp"
+	overlay_state_active = "module_clamp_on"
+	/// Time it takes to load a crate.
+	var/load_time = 3 SECONDS
+	/// The max amount of crates you can carry.
+	var/max_crates = 3
+	/// The crates stored in the module.
+	var/list/stored_crates = list()
+
+/obj/item/mod/module/clamp/on_select_use(atom/target)
+	. = ..()
+	if(!.)
+		return
+	if(!mod.wearer.Adjacent(target))
+		return
+	if(istype(target, /obj/structure/closet/crate))
+		var/atom/movable/picked_crate = target
+		if(!check_crate_pickup(picked_crate))
+			return
+		playsound(src, 'sound/mecha/hydraulic.ogg', 25, TRUE)
+		if(!do_after(mod.wearer, load_time, target = target))
+			return
+		if(!check_crate_pickup(picked_crate))
+			return
+		stored_crates += picked_crate
+		picked_crate.forceMove(src)
+		drain_power(use_power_cost)
+	else if(length(stored_crates))
+		var/turf/target_turf = get_turf(target)
+		if(target_turf.density)
+			return
+		playsound(src, 'sound/mecha/hydraulic.ogg', 25, TRUE)
+		if(!do_after(mod.wearer, load_time, target = target))
+			return
+		if(target_turf.density)
+			return
+		var/atom/movable/dropped_crate = pop(stored_crates)
+		dropped_crate.forceMove(target_turf)
+		drain_power(use_power_cost)
+	else
+		to_chat(mod.wearer, "<span class='warning'>Invalid target!</span>")
+
+/obj/item/mod/module/clamp/on_suit_deactivation(deleting = FALSE)
+	if(deleting)
+		return
+	for(var/atom/movable/crate as anything in stored_crates)
```

Comment:
why `/atom/movable` instead of a more specific type?

Reply:
Good question. Will specify, not sure why TG did it that way, funnily enough gorrilas do too